### PR TITLE
refactor(api): remove model and instrument offsets

### LINKED ISSUE #6799: 4.0.0 Cleanup: remove model offset

### LINKED ISSUE #6801: 4.0.0 Cleanup: Remove instrument offsets from robot config

### PR DESCRIPTION
**Pull request by ahiuchingau**
# Overview
This PR removes the obsolete model offset and instrument offset from Pipette Config and Robot Config respectively. 

closes #6799, closes #6801

# Review Request
I also removed instrumentCalibration from the `calibration/status` endpoint since we don't need it anymore now that instrument offset is removed - is that the right thing to do here?

# Risk assessment
Low, the deleted code is unused since 4.0.0

### CODE REVIEW

**Review by mcous — COMMENTED** [review]
I haven't looked at the code in detail, but removing keys from an HTTP response that a client on an older version may be expecting would be considered a **breaking change** given current HTTP server functionality.

This could be made a non-breaking change if the server was able to recognize the client was on an older HAPI version and migrate the response downwards by dynamically adding the key back in, but given that we don't yet have a solid mechanism for doing that, I think **it would be safer to leave the `instrument_offset` key in the response with meaningless data** (e.g. all zeros, if that makes any sort of sense) with a deprecation TODO or something similar in code

Additional context: https://coda.io/d/HTTP-API-v2_dS1dJrSYETc/API-Versioning_suDJA#_lulfm

**Comment by ahiuchingau** [comment]
@mcous I totally forgot about backwards compatibility when I made the changes. I'll put a TODO and add the key back in

File: api/tests/opentrons/api/test_calibration.py [hunk 1]
```diff
@@ -14,19 +14,10 @@
 
 @pytest.mark.api2_only  # noqa(C901)
 async def test_tip_probe_v2(main_router, model, monkeypatch):
```
Comment (sfoster1) [inline api/tests/opentrons/api/test_calibration.py:16]: can probably just remove this function entirely unless you w ant to do it in a later one

**Review by sfoster1 — APPROVED** [review]
lgtm!

File: api/tests/opentrons/api/test_calibration.py (hunk 1 above)
Comment (ahiuchingau) [reply api/tests/opentrons/api/test_calibration.py:16]: I was gonna save it for #6805

**Pull request merged**